src/runtime/substitution.cc

```diff
@@ -2260,6 +2509,13 @@ tl::optional<GraphOptimizeResult>
   return tl::nullopt;
 }
 
+template <>
+tl::optional<GraphOptimizeResultWithMemory>
+    GraphSearchHelper::try_get_cost_from_cache<GraphOptimizeResultWithMemory>(
```

Comment:
Why no caching?

Reply:
Don't disagree, removing some of the templates will come in after #562 when we can have tests so we don't break anything in the course of the refactors